File: fizzbuzz/fizzbuzz.c
#include <stdio.h>
#include <string.h>

void fizzbuzz(int n, char *buf) {
  if (n % 15 == 0)
    strcpy(buf, "FizzBuzz");
  else if (n % 3 == 0)
    strcpy(buf, "Fizz");
  else if (n % 5 == 0)
    strcpy(buf, "Buzz");
  else
    sprintf(buf, "%d", n);
}

int main() {
  for (int i = 1; i <= 100; i++) {
    char buf[10];
    fizzbuzz(i, buf);
    puts(buf);
  }
  return 0;
}

File: fizzbuzz/fizzbuzz.go
package main

import "fmt"
import "strconv"

func fizzbuzz(n int) string {
  if n % 15 == 0 {
    return "FizzBuzz"
  } else if n % 3 == 0 {
    return "Fizz"
  } else if n % 5 == 0 {
    return "Buzz"
  } else {
    return strconv.Itoa(n)
  }
}

func main() {
    for i := 1; i <= 100; i++ {
       fmt.Println(fizzbuzz(i))
    }
}
